LDAR_Sim/testing/unit_testing/test_scheduling/test_survey_planner/test_update_with_latest_survey.py

```diff
@@ -0,0 +1,110 @@
+"""
+------------------------------------------------------------------------------
+Program:     The LDAR Simulator (LDAR-Sim)
+File:        test_update_with_latest_survey.py
+Purpose: Module for testing test_update_with_latest_survey for FollowUpSurveyPlanner
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the MIT License as published
+by the Free Software Foundation, version 3.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+MIT License for more details.
+You should have received a copy of the MIT License
+along with this program.  If not, see <https://opensource.org/licenses/MIT>.
+
+------------------------------------------------------------------------------
+"""
+
+from datetime import date
+from src.virtual_world.sites import Site
+from scheduling.follow_up_survey_planner import FollowUpSurveyPlanner
+from scheduling.surveying_dataclasses import DetectionRecord
+
+
+def test_update_with_latest_survey(mocker):
+    """
+    Function to test the update_with_latest_survey function in FollowUpSurveyPlanner
+    """
+    mocker.patch.object(Site, "__init__", lambda self, *args, **kwargs: setattr(self, "id", 1))
+    detection_record = DetectionRecord(site_id=1, site=mocker, rate_detected=0.1)
+    detect_date = date(2021, 1, 1)
+    follow_up_survey_planner = FollowUpSurveyPlanner(detection_record, detect_date)
+    method_name = "test_method"
+    redund_filter = FollowUpSurveyPlanner.REDUND_FILTER_RECENT
```

Comment:
If you are going to test redund filter recent - Don't put your biggest rate last since there is also a redund filter max and in that case they would both pass